initial-configuration/jenkins/jenkins-docker/jobs/Build and Deploy Microservice/config.xml

```diff
@@ -0,0 +1,106 @@
+          <description>Describe it</description>
+          <trim>false</trim>
+        </hudson.model.StringParameterDefinition>
+        <hudson.model.FileParameterDefinition>
+          <name>.env</name>
+          <description>The .env file needed to run this service. The UUID will be added to the end of this env</description>
+        </hudson.model.FileParameterDefinition>
+      </parameterDefinitions>
+    </hudson.model.ParametersDefinitionProperty>
+  </properties>
+  <scm class="hudson.plugins.git.GitSCM" plugin="git@5.2.0">
+    <configVersion>2</configVersion>
+    <userRemoteConfigs>
+      <hudson.plugins.git.UserRemoteConfig>
+        <url>https://github.com/hms-dbmi/pic-sure-services.git</url>
+      </hudson.plugins.git.UserRemoteConfig>
+    </userRemoteConfigs>
+    <branches>
+      <hudson.plugins.git.BranchSpec>
+        <name>${git_hash}</name>
+      </hudson.plugins.git.BranchSpec>
+    </branches>
+    <doGenerateSubmoduleConfigurations>false</doGenerateSubmoduleConfigurations>
+    <submoduleCfg class="empty-list"/>
+    <extensions/>
+  </scm>
+  <canRoam>true</canRoam>
+  <disabled>false</disabled>
+  <blockBuildWhenDownstreamBuilding>false</blockBuildWhenDownstreamBuilding>
+  <blockBuildWhenUpstreamBuilding>false</blockBuildWhenUpstreamBuilding>
+  <triggers/>
+  <concurrentBuild>false</concurrentBuild>
+  <builders>
+    <hudson.tasks.Shell>
+      <command># Get the resource from the db if it exists
+export SQL=&quot;SELECT LOWER(CONCAT(SUBSTR(HEX(uuid), 1, 8), &apos;-&apos;, SUBSTR(HEX(uuid), 9, 4), &apos;-&apos;, SUBSTR(HEX(uuid), 13, 4), &apos;-&apos;, SUBSTR(HEX(uuid), 17, 4), &apos;-&apos;, SUBSTR(HEX(uuid), 21))) from picsure.resource where name = &apos;$service_name&apos;&quot;;
+export resource_uuid=$(docker run -i -v /root/.my.cnf:/root/.my.cnf --network=host mysql mysql -se &quot;$SQL&quot; picsure);
+
+# Add the resource to the database if it doesn&apos;t already exist
+if [ -z &quot;$resource_uuid&quot; ]; then
+	echo &apos;This is the first time building this resource. Adding to db&apos;
+    echo &apos;&apos;
+	export SQL=&quot;INSERT IGNORE INTO picsure.resource (uuid, name, resourceRSPath, description) \
+	VALUES (UUID(), &apos;$service_name&apos;, &apos;http://$service_name/&apos;, &apos;$service_description&apos;)&quot;;
+	docker run -i -v /root/.my.cnf:/root/.my.cnf --network=host mysql mysql -e &quot;$SQL&quot; picsure
+fi
+
+# Get the resource from the db 
+export SQL=&quot;SELECT LOWER(CONCAT(SUBSTR(HEX(uuid), 1, 8), &apos;-&apos;, SUBSTR(HEX(uuid), 9, 4), &apos;-&apos;, SUBSTR(HEX(uuid), 13, 4), &apos;-&apos;, SUBSTR(HEX(uuid), 17, 4), &apos;-&apos;, SUBSTR(HEX(uuid), 21))) from picsure.resource where name = &apos;$service_name&apos;&quot;;
+export resource_uuid=$(docker run -i -v /root/.my.cnf:/root/.my.cnf --network=host mysql mysql -se &quot;$SQL&quot; picsure);
+echo &apos;&apos;
+echo &quot;Done adding to db. Using $resource_uuid as uuid&quot;; 
```

Comment:
You could put these in the if statement above, so it skips it if it already exists before.